### PR TITLE
[WIP] Java package name

### PR DESCRIPTION
**Pull request by hsyed**
Hey was thinking of getting into your code generation framework and adding java support so that the validator can generate java validations.

### CODE REVIEW

File: package.go [hunk 1]
```diff
@@ -38,6 +43,7 @@ type pkg struct {
 
 func (p *pkg) ProtoName() Name    { return Name(p.fd.GetPackage()) }
 func (p *pkg) GoName() Name       { return Name(p.name) }
+func (p *pkg) JavaName() Name     { return Name(p.fd.GetOptions().GetJavaPackage()) }
```
Comment (hsyed) [inline package.go]: This should probably return the default package name if `p.fd.GetOptions().GetJavaPackage()` returns zero  -- WDYT ?

File: package.go [hunk 2]
```diff
@@ -16,6 +18,9 @@ type Package interface {
 	// The name of the Go package. This is guaranteed to be unique.
 	GoName() Name
 
+	// JavaName gets the java package as defined in the proto FileDescriptorProto.FileOptions.
+	JavaName() Name
+
```
Comment (hsyed) [inline package.go]: There are some other options for Java -- OuterClassName, MultiClass etc. They probably belong here -- WDYT ?

**Comment by rodaine** [comment]
Thanks for the effort here! I'd recommend holding off on this until the fix for #28 lands. We'll want to move this out into a subpackage as helper functions that operate against `pgs.Node`

**Comment by rodaine** [comment]
Hey! #28 landed, so going to close this PR for now since there should be a separate subpackage for language-specific code (similar to Go now). Going to close this for now; feel free to open it and I'll review!

**Pull request closed without merging**